### PR TITLE
Goodbye Newtonsoft.Json

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

**Comment by bording** [comment]
@danielmarbach I pushed up some tweaks and fixed the failing test.

**Comment by bording** [comment]
For the stuff that's been removed in this PR, is there anything that makes sense to move over to https://github.com/Particular/NServiceBus.Newtonsoft.Json?

**Comment by danielmarbach** [comment]
@bording yeah my plan was to make a comparison tomorrow and then see what could be moved and needs to be tested

File: src/NServiceBus.AcceptanceTests/DataBus/When_sending_databus_properties_with_unobtrusive.cs [hunk 1]
```diff
@@ -64,7 +65,7 @@ public Receiver()
 
                     var basePath = Path.Combine(TestContext.CurrentContext.TestDirectory, @"databus\sender");
                     builder.UseDataBus<FileShareDataBus>().BasePath(basePath);
-                    builder.UseSerialization<JsonSerializer>();
+                    builder.RegisterComponents(c => c.ConfigureComponent<Mutator>(DependencyLifecycle.InstancePerCall));
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/DataBus/When_sending_databus_properties_with_unobtrusive.cs]: shouldn't this use the `RegisterMessageMutator` API?

**Comment by danielmarbach** [comment]
@timbussmann done

**Pull request merged**